client/src/components/SideNav.tsx

```diff
@@ -34,6 +35,10 @@ export const SideNav = ({ open, onClose }: SideNavProps) => {
 
   const [darkMode] = useDarkMode();
 
+  useEffect(() => {
+    document.body.style.overflow = open ? 'hidden' : 'auto';
```

Comment:
this looks haram but apparently its ok for React conventions
 https://stackoverflow.com/questions/39962757/prevent-scrolling-using-css-on-react-rendered-components